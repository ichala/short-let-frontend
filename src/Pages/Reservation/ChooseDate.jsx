import React, { useState } from 'react';
import { checkDate } from '../../Api/reservation/reservationApi';
import styles from './reservation.module.css';

function ChooseDate({ setAvailableHalls, setDate, date }) {
  const [error, setError] = useState(null);
  const [loading, setloading] = useState(false);
  return (
    <section className={`vh-75 mt-5 ${styles.fade_in}`}>
      <div className="container py-5 mt-5 h-100">
        <div className="row d-flex align-items-center justify-content-center h-100 mt-5">
          <div className="col-md-12 col-lg-12 col-xl-12 mt-5">
            <div className={`${styles.card_bg} card text-white mt-md-5`}>
              <div className="card-body">
                <figure className="text-center text-sm">
                  <blockquote className="blockquote text-center">
                    <h3 className="display-4">Choose a date</h3>
                  </blockquote>
                  <figcaption className="blockquote-footer text-white">
                    <cite className="text-white">
                      Choose a date to see available halls
                    </cite>
                  </figcaption>
                </figure>
                <form
                  onSubmit={(e) => {
                    e.preventDefault();
                    checkDate(setError, setAvailableHalls, date, setloading);
                  }}
                  className="flex-nowrap d-flex justify-content-center align-items-center flex-column "
                >
                  <p className="text-danger mb-1 ">
                    {error && error.message}
                  </p>
                  <input
                    type="date"
                    className="form-control "
                    required
                    onChange={(e) => {
                      setError(null);
                      setDate(e.target.value);
                    }}
                  />
                  {loading ? (
                    <div
                      className="spinner-grow mt-3 text-md text-center text-success"
                      role="status"
                    />
                  ) : (
                    <button type="submit" className="btn btn-dark mt-3">
                      Next
                    </button>
                  )}
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
}

export default ChooseDate;
