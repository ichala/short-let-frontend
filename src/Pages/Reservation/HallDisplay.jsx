import React from 'react';
import styles from './reservation.module.css';

function HallDisplay({ hall }) {
  return (
    <>

      <div className={`card ${styles.fade_in} mb-3 mt-4 ${styles.hall_card}`}>
        <div className="row g-0">
          <div className="col-md-4">
            <img src={hall.image} className={`img-fluid rounded-start ${styles.hall_image}`} alt={hall.name} />
          </div>
          <div className="col-md-8">
            <div className="card-body">
              <h5 className={`card-title display-6 text-white font-bold ${styles.text_main}`}>{hall.name}</h5>
              <p className="card-text">{hall.description}</p>
              <p className="card-text display-6 ">
                {Number(hall.cost)}
                $/Day
              </p>
            </div>
          </div>
        </div>
      </div>

    </>
  );
}

export default HallDisplay;
