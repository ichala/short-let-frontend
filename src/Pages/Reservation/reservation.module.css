.card_bg {
  background: var(--main);
}

input,
select {
  max-width: 280px;
}

.text_main {
  color: var(--text-primary);
}

.fade_in { animation: fadeIn 1.5s; }

@keyframes fadeIn {
  0% { opacity: 0; }
  100% { opacity: 1; }
}

.hall_image {
  height: 300px;
}

.hall_card {
  min-height: 300px;
  min-width: 100%;
  background-color: rgba(63, 98, 18, 0.6);
}

.button {
  width: 180px;
  background-color: #a3e635;
}
