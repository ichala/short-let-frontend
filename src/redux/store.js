import { configureStore } from '@reduxjs/toolkit';
import authSlices from './slices/auth';

export default configureStore({
  reducer: {
    user: authSlices,
  },
});
