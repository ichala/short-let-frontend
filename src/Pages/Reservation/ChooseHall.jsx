import React, { useState } from 'react';
import Swal from 'sweetalert2';
import { BiArrowBack } from 'react-icons/bi';
import { PostReservation } from '../../Api/reservation/reservationApi';
import HallDisplay from './HallDisplay';
import styles from './reservation.module.css';

function ChooseHall({
  setAvailableHalls, availableHalls, date, setSaved,
}) {
  const [error, setError] = useState(null);
  const [loading, setloading] = useState(false);
  const [hall, setHall] = useState(availableHalls[0]);
  function ChooseHall(id) {
    availableHalls.forEach((hall) => {
      if (hall.id === Number(id)) {
        setHall(hall);
      }
    });
  }
  function ConfirmReservation() {
    Swal.fire({
      title: `Do you want to confirm the reservation ? <br> <p class="mt-2 display-5"> ${hall.name} </p> <br><p class="display-6">  Cost: ${Number(hall.cost)}$ </p>`,
      showDenyButton: true,
      confirmButtonText: 'Confirm',
      denyButtonText: 'Cancel',
      allowOutsideClick: false,
    }).then((result) => {
      /* Read more about isConfirmed, isDenied below */
      if (result.isConfirmed) {
        PostReservation(date, hall).then((res) => {
          if (res.status === 200) {
            Swal.fire('Reservation Request Submitted', '', 'success');
            setSaved(true);
          }
        }).catch((e) => {
          if (e.toJSON().message === 'Network Error') {
            Swal.fire('No Internet Or Server is not running', '', 'danger');
            setloading(false);
          } else {
            Swal.fire(e.response.data.message, '', 'info');
            setloading(false);
          }
        });
      } else if (result.isDenied) {
        Swal.fire('Reservation Canceled', '', 'info');
      }
    });
  }

  return (
    <section className="vh-75 m-auto">
      <div className="container mt-5 h-100">
        <div className="row d-flex align-items-center justify-content-center h-100">
          <div className="col-md-12 col-lg-12 col-xl-12">
            <button
              onClick={() => setAvailableHalls([])}
              type="button"
              className="btn btn-success mb-2"
            >
              <BiArrowBack size="30" />
            </button>
            <div className={`${styles.card_bg} card text-white mt-md-3`}>
              <div className="card-body">
                <figure className="text-center text-sm">
                  <blockquote className="blockquote text-center">
                    <h3 className="display-4">Choose a Hall</h3>
                  </blockquote>
                  <figcaption className="blockquote-footer text-white">
                    <cite className="text-white">
                      Choose your date and we will list the available halls for
                      you
                    </cite>
                  </figcaption>
                </figure>
                <form
                  onSubmit={(e) => {
                    e.preventDefault();
                    ConfirmReservation();
                  }}
                  className="flex-nowrap d-flex justify-content-center align-items-center flex-column "
                >
                  <p className="text-danger mb-1 ">{error && error.message}</p>
                  <select
                    disabled={loading}
                    className="form-select text-center"
                    aria-label="Default select example"
                    required
                    onChange={(e) => {
                      setError(null);
                      ChooseHall(e.target.value);
                    }}
                  >
                    {availableHalls.map((hall) => (
                      <option key={hall.id} value={hall.id}>
                        {hall.name}
                      </option>
                    ))}
                  </select>

                  <HallDisplay hall={hall} />
                  {loading ? (
                    <div
                      className="spinner-grow mt-3 text-md text-center text-success"
                      role="status"
                    />
                  ) : (
                    <div className="font-weight-bold">
                      <button type="submit" className="btn btn-dark mt-3">
                        Book Now
                      </button>
                    </div>
                  )}
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
}

export default ChooseHall;
