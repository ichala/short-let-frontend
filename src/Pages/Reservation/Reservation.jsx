import { useState } from 'react';
import { Navigate } from 'react-router-dom';
import ChooseDate from './ChooseDate';
import ChooseHall from './ChooseHall';

function Reservation() {
  const [date, setDate] = useState('');
  const [availableHalls, setAvailableHalls] = useState([]);
  const [Saved, setSaved] = useState(null);

  if (Saved) {
    return <Navigate to="/my-reservations" replace />;
  }

  if (availableHalls.length > 0) {
    return (
      <ChooseHall
        date={date}
        setSaved={setSaved}
        availableHalls={availableHalls}
        setAvailableHalls={setAvailableHalls}
      />
    );
  }

  return (
    <>
      <ChooseDate
        date={date}
        setDate={setDate}
        setAvailableHalls={setAvailableHalls}
      />
    </>
  );
}

export default Reservation;
