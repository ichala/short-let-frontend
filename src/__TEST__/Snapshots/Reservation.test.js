import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../redux/store';
import Reservation from '../../Pages/Reservation/Reservation';

describe('Login', () => {
  it('should render the login page', () => {
    const app = render(
      <Provider store={store}>
        <Router>
          <Reservation />
        </Router>
      </Provider>,
    );
    expect(app).toMatchSnapshot();
  });
});
