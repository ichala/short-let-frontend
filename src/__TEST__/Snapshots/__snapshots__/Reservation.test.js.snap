// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the login page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="vh-75 mt-5 fade_in"
      >
        <div
          class="container py-5 mt-5 h-100"
        >
          <div
            class="row d-flex align-items-center justify-content-center h-100 mt-5"
          >
            <div
              class="col-md-12 col-lg-12 col-xl-12 mt-5"
            >
              <div
                class="card_bg card text-white mt-md-5"
              >
                <div
                  class="card-body"
                >
                  <figure
                    class="text-center text-sm"
                  >
                    <blockquote
                      class="blockquote text-center"
                    >
                      <h3
                        class="display-4"
                      >
                        Choose a date
                      </h3>
                    </blockquote>
                    <figcaption
                      class="blockquote-footer text-white"
                    >
                      <cite
                        class="text-white"
                      >
                        Choose a date to see available halls
                      </cite>
                    </figcaption>
                  </figure>
                  <form
                    class="flex-nowrap d-flex justify-content-center align-items-center flex-column "
                  >
                    <p
                      class="text-danger mb-1 "
                    />
                    <input
                      class="form-control "
                      required=""
                      type="date"
                    />
                    <button
                      class="btn btn-dark mt-3"
                      type="submit"
                    >
                      Next
                    </button>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="vh-75 mt-5 fade_in"
    >
      <div
        class="container py-5 mt-5 h-100"
      >
        <div
          class="row d-flex align-items-center justify-content-center h-100 mt-5"
        >
          <div
            class="col-md-12 col-lg-12 col-xl-12 mt-5"
          >
            <div
              class="card_bg card text-white mt-md-5"
            >
              <div
                class="card-body"
              >
                <figure
                  class="text-center text-sm"
                >
                  <blockquote
                    class="blockquote text-center"
                  >
                    <h3
                      class="display-4"
                    >
                      Choose a date
                    </h3>
                  </blockquote>
                  <figcaption
                    class="blockquote-footer text-white"
                  >
                    <cite
                      class="text-white"
                    >
                      Choose a date to see available halls
                    </cite>
                  </figcaption>
                </figure>
                <form
                  class="flex-nowrap d-flex justify-content-center align-items-center flex-column "
                >
                  <p
                    class="text-danger mb-1 "
                  />
                  <input
                    class="form-control "
                    required=""
                    type="date"
                  />
                  <button
                    class="btn btn-dark mt-3"
                    type="submit"
                  >
                    Next
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
